### ISSUE TITLE
Please release the app on F-Droid 

### ISSUE THREAD

**Issue by YamiYukiSenpai** [RikkaApps/Shizuku#69]
Any plans on releasing this app on F-Droid? 

Please and thank you.

**Comment by haruue** [comment]
Sorry, but currently we have no plan to publish this application itself on F-Droid.

**Comment by IzzySoft** [comment]
I was just considering taking your app into [my repo](https://apt.izzysoft.de/fdroid), but then I saw in your Readme:

> You are **FORBIDDEN** to use image files listed below in any way.

The list includes the app icon, which would be shown in the lists (and on the details page) next to the app name. Well, going by the letter, that's using it "in any way". If that's not what you've meant (after all, the app drawer on Android devices would also show it), can you please explicitly state so?

Readme also says:

> You are **FORBIDDEN** to distribute the apk compiled by you … to any store

So if I take the app compiled by YOU, that would be OK? Not sure if you'd count my repo into "any store" (I don't sell things) – or if that would apply at all (I don't compile apps either).

**TL;DR:** Would it be OK with you if I take the APK **you** compiled (and attached to your releases here) and make it available via my repo (including fetching and providing updates within 24h of their being attached here)? And if so, would there be any screenshots you'd like (or permit) to show along with its listing?

And PS: Your "Apps using Shizuku" could be complemented by [SAI](https://github.com/Aefyr/SAI) :wink:

**Comment by RikkaW** [comment]
@IzzySoft Yes, of course.

The original purpose of "FORBIDDEN"s is to prevent people who publish modified opensource projects to Play Store. If the license allows, there are people doing this. And I know there are projects hurt from this.

**Comment by IzzySoft** [comment]
Thanks a lot! I'm now adding your app to my repo (should [show up here](https://apt.izzysoft.de/fdroid/index/apk/moe.shizuku.privileged.api) in about 6 hours (with the next sync). Updates are fetched automatically within 24h of your attaching an APK to a new tag. Be welcome to link there from your [Download page](https://shizuku.rikka.app/download.html) :wink:

**Comment by songsammy** [comment]
On fdroid , all you can do is to submit the source code and fdroid does the complie.
So can you, the author, upload the code to fdroid?
I mean fdroid official repo, so we can search and install from fdroid

**Comment by IzzySoft** [comment]
@songsammy you don't "upload" to F-Droid. You can open a [Request for Packaging](https://gitlab.com/fdroid/rfp/issues) to have an app included. To put it in simple terms, F-Droid would then pull the code from the app's repo to build it. But the following clause of the Readme makes this impossible:

> You are FORBIDDEN to distribute the apk compiled by you (including modified, e.g., rename "Shizuku" to something else) to any store (IBNLT Google Play Store, etc.).

F-Droid needs to compile from the source, and then distributes the resulting APK.

**Comment by songsammy** [comment]
Sorry for the mistake. That's why i am here, requesting the author to release on fdroid

**Comment by linsui** [comment]
Just my two cents here.

>The original purpose of "FORBIDDEN"s is to prevent people who publish modified opensource projects to Play Store. If the license allows, there are people doing this. And I know there are projects hurt from this.

First of all, thanks for your app! I understand your concern, but I doubt how effective it is. Generally a misbehaving developer won't care about it but just steal your code. However this will prevent F-Droid build your app from source and distribute it. Also, I wonder this exception will make the app non-FOSS but I'm not a lawyer.

**Comment by EmanuelLoos** [comment]
> Also, I wonder this exception will make the app non-FOSS but I'm not a lawyer.

I'm not a lawyer either but I do know quite a lot about free software and software licenses and I am 99% sure this goes against freedom 2 & 3 of [the four essential freedoms of free software defined in the Free Software Definition](https://www.gnu.org/philosophy/free-sw.html) if it is legally valid which is not clear to me because Shizuku is licensed under the terms of the Apache-2.0 License.

> The original purpose of "FORBIDDEN"s is to prevent people who publish modified opensource projects to Play Store. If the license allows, there are people doing this. And I know there are projects hurt from this.

You are right this is a problem with permissive free software licenses like the Apache-2.0 License which is solved by copyleft licenses for free software like the [GNU General Public License (GPL)](https://en.wikipedia.org/wiki/GNU_General_Public_License). These licenses introduce conditions on distributing the code that are not taking away any of the mentioned freedoms but requiring anyone using the code to comply with them. Specifically the GPL requires anyone who wishes to distribute your app to make the source code of his/her version available under the terms of this license as well so that you can use any of these modifications in our app as well. Both the GPL and the Apache-2.0 License require attribution. Anyone can take code from your project but they need to mention it is your work, not theirs. The GPL is a bit more strict here as section 5d of the GPL states:

    d) If the work has interactive user interfaces, each must display
    Appropriate Legal Notices; however, if the Program has interactive
    interfaces that do not display Appropriate Legal Notices, your
    work need not make them do so.

In relation to this app this means the UI must inform the user who worked on code used in this app ("copyright owners") and display the license of each part of the app. Normally this is done in the "About" section. For example you may take a look at how [NewPipe](https://github.com/TeamNewPipe/NewPipe) which is licensed under the terms of the GPLv3-or-later, this way it can easily switch to a newer version of the GPL. Copyleft is quite difficult so to avoid code needing to be rewritten when switching to a newer version of the GPL this optional additional permission is granted, which I would recommend as well.

[The GPL, version 3 is compatible with the Apeache-2.0 License which means, if you like, you can simply switch from the Apache-2.0 License to the GPL, version 3.](https://www.gnu.org/licenses/license-list.html#apache2)

[There is a GPL howto which can help you with licensing if you choose the GPL.](https://www.gnu.org/licenses/gpl-howto.html)

**Comment by IzzySoft** [comment]
Yes, [reproducible builds](https://reproducible-builds.org/) are a good thing:

* [Reproducible Builds](https://f-droid.org/docs/Reproducible_Builds/)
* [Our build and release infrastructure, and upcoming updates](https://f-droid.org/2022/05/24/buildserver-overhaul-sponsored-by-calyx-institute.html)
* [Trust is good, control is better](https://nico.dorfbrunnen.eu/posts/2019/reproducibility-fdroid/)

:wink:

**Comment by yashpalgoyal1304** [comment]
https://github.com/RikkaApps/Shizuku/blob/234b1c8335e821e63fd5a4d923627b358ccfe11e/README.md?plain=1#L101

yeah, i was also thinking that if distributing self compiled apk is forbidden, then reproducible builds can be used. so, if the build by fdroid matches with the apk released here, this one is distributed, otherwise, that build/version/tag is dropped.

but yeah, i am just thinking, not saying that this should be done forcefully or so...

**Comment by Mannshoch** [comment]
I requested this app in fdroid forum and got a link to this discussion:
https://forum.f-droid.org/t/request-shizuku/21957/1

**Comment by shaggyskunk** [comment]
You know that Shizuku is available on apk mirror.

**Comment by Mannshoch** [comment]
APK Mirror is mor worse than downloading it from github or fdroid. I do not know the people or organisation behind that Website. And I'm not able to compare any hash.

**Comment by shaggyskunk** [comment]
apk mirror is run by Android Police & is exceptionally safe.
If the Dev does not want to release it on F-Droid - That's entirely up to
him.
I've given you an alternative - If you don't like it - Too bad.

**Comment by Mannshoch** [comment]
I do not know what you mean by _run by Android Policy_ I'm sure this website is not from Google.

The app from RikkaApps is trustworthy by the widely using and - I hope so - some more eye that look into the code. And Fdroid because I personally misuse them as a verification of the Code. 
I know they do not look deep into the code and do not promise anything but I use what ever I get.

**Comment by haruue** [comment]
> And I'm not able to compare any hash.

@Mannshoch 

Android has a built-in mechanism called package signature verification that can reliably check the hash for you.

Ensure that you are not utilizing any Xposed modules, such as Core Patch, to disable package signature verification. Download Shizuku from the GitHub release for the first time, and thereafter, you can download it from any source of your choice (including APK Mirror). The package signature verification will take care of everything to ensure that you install an authentic version of Shizuku.

**Comment by linsui** [comment]
You can get the apk from the release directly. No need to use apk mirror. If you want update check, you can get it from IzzySoft's F-Droid repo.

**Comment by shaggyskunk** [comment]
Way too much fuss over this.. 🙄

**Comment by IzzySoft** [comment]
Well, here's a compromise: Enable the [IzzyOnDroid repo](https://apt.izzysoft.de/) and find [Shizuku here](https://apt.izzysoft.de/packages/moe.shizuku.privileged.api/). With the F-Droid app. Including automated updates. And more than 1.000 other apps. As linsui just suggested :wink:

**Comment by shuvashish76** [comment]
> _F-Droid modifies the source code at build time, you should not trust F-Droid, GitHub Actions is a verifiable and reproducible build._ (comment)

@linsui Reproduciable builds possible for F-Droid? If yes then it shouldn't be an issue for Shizuku devs. 🧐

**Comment by linsui** [comment]
It's possible.

**Comment by IzzySoft** [comment]
@RikkaW just a minor point that could also help (we had this "is-it-foss" discussions on many projects using your library, which luckily is not affected by that phrase):

> The original purpose of "FORBIDDEN"s is to prevent people who publish modified opensource projects to Play Store. If the license allows, there are people doing this. And I know there are projects hurt from this.

Do you really think those copycats would care for that clause (also see [this comment](#issuecomment-795289619))? I think what you mean by "modified" here is "ad-and-tracking-enriched" and not even giving credits. There was just one such copycat taken down entirely from Play after multiple of the original authors complained to Google (one that I'm aware of that is), so there'd be means to deal with that.

> F-Droid modifies the source code at build time

What gives you that idea? That's usually not the case. And using [reproducible builds](https://gitlab.com/-/snippets/2544837) (as it was suggested) will even ensure that's not the case, and can be verified.

So would you consider? Not pressing, just suggesting :wink:

**Comment by EmanuelLoos** [comment]
> We don't care if the app is FOSS or not, we expose the code for the purpose of sharing knowledge. Allow anyone to use our code, include closed source software (GPL takes away the right of closed source software to use our code). However, distribution of modified apks is not welcome, which can make it difficult to handle feedback.

You nicht also want to consider dual licensing, like so:

Give anyone who wants to use the code  the choice between two licences:
1. the FORBIDDENs, but the code can also be used in non-free Software
2. the GPLv3-or-later

That way you could share knowledge with all software developers while effectively avoiding copycats and making the app eligible for F-Droid (they would still wait for your specific agreement before doing anything with your app), you can inform them about your requirement of reproducible builds and I think they would be more than happy to meet that requirement, because AFAIK they are striving for reproducible builds as well.

**Comment by Meow-purr** [comment]
> > We don't care if the app is FOSS or not, we expose the code for the purpose of sharing knowledge. Allow anyone to use our code, include closed source software (GPL takes away the right of closed source software to use our code). However, distribution of modified apks is not welcome, which can make it difficult to handle feedback.
> 
> You might also want to consider dual licensing, like so:
> 
> Give anyone who wants to use the code the choice between two licences:
> 
>     1. the FORBIDDENs, where the code can also be used in non-free software, or
> 
>     2. the GPLv3-or-later
> 
> 
> That way you could share knowledge with all software developers (free and non-free) while avoiding copycats just as effectively and making the app eligible for F-Droid (they would still wait for your specific agreement before doing anything with your app and won't touch it if you say no), you can inform them about your requirement of reproducible builds and I think they would be more than happy to meet that requirement, because they are striving for [reproducible builds](https://f-droid.org/docs/Reproducible_Builds/) as well, in fact, they are currently headed [towards a reproducible F-Droid](https://f-droid.org/docs/Reproducible_Builds/), are ["encouraging new apps to enable reproducible builds"](https://f-droid.org/docs/Reproducible_Builds/) and [have "gotten many more reproducible apps in F-Droid than before" lately](https://gitlab.com/fdroid/fdroiddata/-/issues/2844).
> 
> If you like that idea, I can also provide assistance with licensing and making the (most important) conditions clear in the README.

i think that gets a +1

**Comment by alexgieg** [comment]
I'd like to point out that the "Forbidden" has no meaning at all for any code that says it's Apache 2.0 licensed. If a file says it's Apache 2.0 licensed, it's Apache 2.0 licensed, including the right to recompile and distribute. This is the case irrespective of the README saying the project as a whole doesn't actually comply with the Apache 2.0 License. If someone were to fork the entire master branch, then delete the README file and any file that isn't licensed by the Apache 2.0 License, they'd end up with a resulting branch that's entirely made of Apache 2.0 Licensed files, which thus explicitly allows redistribution of recompiled binaries.

To make the code non-redistributable in recompiled form, the project as a whole, meaning all of its individual files that aren't 3rd party libraries, would need to be relicensed either under a license crafted so as to forbid redistributing recompiled versions or, at the very minimum, the Apache 2.0 License should be removed from the repository and from the files. This would mark the project, from GitHub's perspective, as not having a general license, that is, as being fully copyright-restricted under whatever terms the author chose. The only things that are allowed in this case, per GitHub's Terms of Service, is for the code to be viewable and forkable -- which is what you seem to want it to be. I'd suggest doing that, plus adding a freeware EULA.

Notice that making the code proprietary may cause it not to be legally allowed to use certain libraries that are only available to fully open-source (OSI definition) or free-software (FSF definitions) software, so you'd need to look into the libraries you use and replace the ones that are incompatible with usage within proprietary software (even if it's freeware), if any.

**Comment by vvb2060** [comment]
Android applications differ from Linux programs in that they have package names and signatures. The package name is equivalent to a domain name, which should be unique in the world and show the owner, and the signature is equivalent to a TLS certificate for the domain name. 
We authorize everyone to use shizuku code (Java and Kotlin) because we believe it is enough for developers. If someone wants to use the package name, icon and app name of shizuku (Gradle, PNG and XML), then this is a counterfeit website and we cannot be responsible for its content.
We don't want to care about FOSS, we don't want to care about license, because they are too much of a hassle, and if someone does do something, there is nothing we can do about it. We just think that apps shouldn't exec su and then parse the return string to IPC, calling privileged APIs should be like normal APIs, so made this app and share it with you. If your app needs to call privileged APIs and you don't like shizuku, you can include the whole code in your app and do it all yourself.

**Comment by linsui** [comment]
IIUC, anyone who want to include Shizuku in F-Droid should fork this repo and change the app name, id and icon, and make it clear that it's a fork of Shizuku. Then please open an MR to fdroiddata.

**Comment by Meow-purr** [comment]
@vvb2060 than just use Apache and have the icons and name and app id as trademarks to solve disputes

**Comment by rokejulianlockhart** [comment]
> Android applications differ from Linux programs in that they have package names and signatures.

@vvb2060, any vaguely decent package manager supports AppStream and package signatures. That's wrong.

**Comment by Pandapip1** [comment]
This might be beating a dead horse, but instead of this weird dual license with exceptions idea, why not just use the LGPL? Shizuku forks would have to be LGPL-licensed, but proprietary apps can use Shizuku just fine.

**Comment by Atemu** [comment]
Before coming across this issue, I have opened https://github.com/RikkaApps/Shizuku/discussions/359. That may perhaps be a better place to discuss this topic going forward.

A couple notes on this issue's discussion:

* The terms added to the README are part of the license. It does not much matter whether they're stated in the LICENSE file or README file. Though obviously it'd be clearer for everyone if they were also put into the LICENSE file.
* In my understanding, sublicensing allows you to place further restrictions on a license but not to remove existing terms. You cannot remove the distribution restriction via sublicensing; you'd have to re-license.

**Comment by eternal-sorrow** [comment]
Why is this closed? Until the solution is found and the app is published on F-droid, this issue should remain open.

**Comment by Atemu** [comment]
Unless the license changes (which I don't see any intention of happening any time soon sadly), this is effectively a WONTFIX.

There is no point in tracking issues that the developer explicitly does not intend to fix.

**Comment by eternal-sorrow** [comment]
>Unless the license changes (which I don't see any intention of happening any time soon sadly)

Which means the developer is not trustworthy and the app should not be used.

**Comment by NerioAlvaro** [comment]
I totally agree.

**Comment by shuvashish76** [comment]
@linsui https://github.com/RikkaApps/Shizuku/commit/7bd50a24189446114d2fd669842bbdc720411142 now it's FOSS-compliant. @IzzySoft Remove the Anti-Feature from IoD.

**Comment by eternal-sorrow** [comment]
Question: If someone creates a third party build that is compliant to those requirements, i.e., uses a different app id and a different permission name, will the other apps that use Shizuku API be able to use that build without modification?

**Comment by eternal-sorrow** [comment]
You know it's unacceptable for places like F-Droid.

**Comment by Pandapip1** [comment]
If it's truly bit for bit reproducible, it shouldn't be a problem for F-Droid to just compile it themselves and distribute that, since there is no difference between the builds?

**Comment by Pandapip1** [comment]
> It's not clear if this is still Apache-2.0 license..

I interpreted the clauses as meaning that Shizuku, the Shizuku branding, the app ID, and its permissions were being claimed to be common law trademarks. In this regard, it is still Apache 2.0, even if it means that Shizuku is (still) unfree software (yes, it's possible to use an OSI-approved license in a way that makes the program as a whole unfree).

**Comment by shuvashish76** [comment]
Google Play Store: They really don't care about your license, you've to fight for copycats with little hope for removal.
Amazon Appstore: They already [announced](https://www.theverge.com/news/616646/amazon-appstore-ending-android-support) to close down the Appstore.
F-Droid: It's [necessary to have unique AppID](https://f-droid.org/en/docs/Inclusion_Policy/#:~:text=The%20software%20should,%20guidance).
Others: Same issue as Google with their own policy.

How useful are those special clauses?

**Comment by vvb2060** [comment]
buildid is useful debug tool, so we will not disable it. I've compared the output of CI builds:  https://github.com/RikkaApps/Shizuku/actions/runs/15236707222  https://github.com/RikkaApps/Shizuku/actions/runs/15302750215 , and the sha512 of so file is identical, so it's reproducible.

**Comment by eternal-sorrow** [comment]
I repeat my question: If someone creates a third party build that is compliant to those requirements, i.e., uses a different app id and a different permission name, will the other apps that use Shizuku API be able to use that build without modification? I'm not an Android app developer, so I don't know how this works.

**Comment by IzzySoft** [comment]
> buildid is useful debug tool

For debug builds, OK. But we're talking release builds here.

> so it's reproducible.

Well, deterministic. You can build it multiple times in a row in your environment on Github here and get a byte-identical APK. I can't outside of Github, due to the build-id (I probably get byte-identical APKs in multiple runs here, too, but none that matches yours). As the sha256hash in my case differs, I cannot confirm RB (nor can F-Droid) 🤷‍♂ A good RB is one confirmed by an **_independent_** party.

**Comment by linsui** [comment]
You can try matching the path used in GHA.

**Comment by IzzySoft** [comment]
> You can try matching the path used in GHA.

I did that:

```yaml
        build_home_dir: /home/runner
        build_repo_dir: /home/runner/work/Shizuku/Shizuku
```

It might well be a difference in the `cmake` version, or in some other dependency. But the easiest way to be surely rid of it would be disabling the build-ids for release builds.

**Comment by vvb2060** [comment]
> For debug builds, OK. But we're talking release builds here.

Build-id is used to identify the version in native crash reports. 

> It might well be a difference in the cmake version, or in some other dependency.

I upgraded the cmake version. Can you test new GHA output?

**Comment by IzzySoft** [comment]
> Build-id is used to identify the version in native crash reports.

A commit hash wouldn't do for that? At least that would be deterministic.

> I upgraded the cmake version. Can you test new GHA output?

Which one did you use?  According to `sdkmanager --list`, the ones we can set here are `3.22.1` (which was used in the last build IIRC), `3.30.(3|4|5)`, `3.31.(0|1)` (plus some older ones). Ah, [3.31](https://github.com/RikkaApps/Shizuku/blob/16d28799e4039f0c008ec20513be20174a087069/manager/src/main/jni/CMakeLists.txt#L1). Is there a matching APK available to check against?

**Comment by vvb2060** [comment]
apk in GHA output (Artifacts)

**Comment by IzzySoft** [comment]
Nice, that's RB! I had to set the cmake version explicitly, but it worked out fine. So with the next release, we can establish RB here at IzzyOnDroid 🥳 Can you give me a ping here when that's available?

**Comment by vvb2060** [comment]
I think the next release will be much later (when Google breaks the hidden API again?), maybe it's better to subscribe to release feed.

**Comment by IzzySoft** [comment]
> maybe it's better to subscribe to release feed.

Not an option for me. I'm already drowning in notification mails, so I prefer not to subscribe to feeds here and there and lose it completely, sorry…

**Comment by Integral-Tech** [comment]
According to the project README:
> You are FORBIDDEN to distribute the apk that use Shizuku as name or use moe.shizuku.privileged.api as application id or declare moe.shizuku.manager.permission.* permission to any store (IBNLT Google Play Store, F-Droid, Amazon Appstore etc.).

[The OSI definition](https://opensource.org/osd) says:
> 1. Free Redistribution
The license shall not restrict any party from selling or giving away the software as a component of an aggregate software distribution containing programs from several different sources. The license shall not require a royalty or other fee for such sale.

[The definition of free software](https://www.gnu.org/philosophy/free-sw.en.html#four-freedoms) says:
> The freedom to distribute copies of your modified versions to others (freedom 3). By doing this you can give the whole community a chance to benefit from your changes. Access to the source code is a precondition for this.

Therefore Shizuku is neither open-source software nor free software. It is [source-available software](https://en.wikipedia.org/wiki/Source-available_software).

**Comment by EmanuelLoos** [comment]
@Integral-Tech 

I woluld not be so sure of that. Both free software and open source Software are very clear about trademarks as well. The Apache License is a well-known FLOSS license and the part oft the Readme you are quoting is explcitly refering to section 6 of the Apache License:

>Under Apache 2.0 section 6, specifically:
>
>• You are FORBIDDEN to use manager/src/main/res/mipmap*/ic_launcher*.png image files, unless for displaying Shizuku itself.
>
>• You are FORBIDDEN to use Shizuku as app name or use moe.shizuku.privileged.api as application id or declare moe.shizuku.manager.permission.* permission.

The Shizuku developers are just claiming trademark right for their launcher images as well as their app and permission names. This is common practice for FLOSS projects. Tradmarks can be APP namens and app icons: https://arapackelaw.com/trademarks/trademarks-for-mobile-apps/

"Prohibiting misrepresentation of the origin of that material, or requiring that modified versions of such material be marked in reasonable ways as different from the original version" is allowed even by the GPL.

**Comment by Integral-Tech** [comment]
@EmanuelLoos After I posted my comment, the developer updated the README: https://github.com/RikkaApps/Shizuku/commit/b844bc491f1790c72328e1a8e5b2349f8978f0ea

**Comment by vvb2060** [comment]
Yes, using these names is not allowed in any case now, not just when distributing in app store. I'm surprised you didn't like this right and therefore took back the rights grant.

**Comment by eternal-sorrow** [comment]
I can agree about an image and app name, those can be trademarked of course. But about app id and permission id - I'm not so sure. Those could be crucial for compatibility with existing apps, using the Shizuku API (the developer didn't clarify if it is or not despite me asking twice). In this case those shouldn't be trademarked.

**Comment by Integral-Tech** [comment]
> There are two problems:
> 
>     1. It's not allowed to publish Shizuku as is. "You are FORBIDDEN to distribute the apk that use Shizuku as name or use moe.shizuku.privileged.api as application id or declare moe.shizuku.manager.permission.* permission to any store (IBNLT Google Play Store, F-Droid, Amazon Appstore etc.)."
> 
>     2. It's not clear if this is still Apache-2.0 license..

@linsui The project README has been updated: https://github.com/RikkaApps/Shizuku/commit/b844bc491f1790c72328e1a8e5b2349f8978f0ea

Is it possible to publish it to F-Droid using reproducible build now?

**Comment by linsui** [comment]
> You are **FORBIDDEN** to use `Shizuku` as app name or use `moe.shizuku.privileged.api` as application id or declare `moe.shizuku.manager.permission.*` permission.

How can we build it if we can't use the name, appid and permission?

**Comment by EmanuelLoos** [comment]
@linsui 

We can build it and compare the output to the official binaries. If they match, we can redistribute the official binaries, because they are exactly the same as our build.

Although you are right that additional clarification would be best. Something like:

"You are FORBIDDEN to use Shizuku as app name or use moe.shizuku.privileged.api as application id or declare moe.shizuku.manager.permission.* permission, except when redistributing our official binaries."

**Comment by Integral-Tech** [comment]
@linsui @EmanuelLoos I've just asked the developer, and he confirmed that open-source licenses have no impact on the distribution of prebuilt APKs. Therefore, it is unnecessary to emphasize "except when redistributing our official binaries".

**Comment by IzzySoft** [comment]
Quoting from [The Four Essential Freedoms of Free Software](https://en.wikipedia.org/wiki/The_Free_Software_Definition#The_Four_Essential_Freedoms):

> The freedom to study how the program works, and change it to make it do what you wish (freedom 1).
>
> The freedom to distribute copies of your modified versions to others (freedom 3).

How does that work then? OK, one can rename the app, and give it a different AppId. But what use is that without the permission? It would be incompatible at best, as the apps using the API would (need to) request exactly that permission. So freedom-1 and freedom-3 are at least limited if the permission name cannot be used.

**Comment by CoelacanthusHex** [comment]
And in law, if a trademark is used in compatibility check, it won't be protected in this case. See *Sega Enterprises Ltd. v. Accolade, Inc.* (977 F.2d 1510 (9th Cir. 1992)).

> You are FORBIDDEN to use Shizuku as app name or use moe.shizuku.privileged.api as application id or declare moe.shizuku.manager.permission.* permission.

If F-Droid uses application id and permission name for the purpose of compatibility with existing software, the trademark protection will not be legally effective.

**Comment by shuvashish76** [comment]
From @vvb2060 's telegram channel: https://t.me/vvb2060Channel/1062

> https://f-droid.org/zh_Hans/2025/09/29/google-developer-registration-decree.html
> 
>  It's a bit funny. F-Droid not only can but also has the responsibility to register with Google on behalf of developers, because the policy states that the holder of the signing certificate private key should register their own identity, not the app developer, although in most cases they are the same. If F-Droid chooses to distribute APKs signed with the app developer's private key, then the registration responsibility lies with the developer. Although F-Droid claims that if the build is reproducible, it will use the original developer's private key for distribution, and F-Droid says it cannot take over the application identifier of the open-source apps it distributes because that would effectively seize exclusive distribution rights of those apps. But in practice, the attitude is completely opposite: https://github.com/RikkaApps/Shizuku/issues/69 Due to F-Droid's policy requirements, I tried to make Shizuku support reproducible builds, but it was still rejected by F-Droid for APK distribution because Shizuku prohibits others from using the current application identity according to Apache 2.0 Section 6, so Shizuku is not free software.

<details><summary>Original text:</summary>
<p>

> https://f-droid.org/zh_Hans/2025/09/29/google-developer-registration-decree.html
> 
> 有点搞笑。F-Droid不仅可以并且本来就有责任代表开发者向Google注册，因为政策是签名证书私钥持有者应该注册自己的身份，而不是应用开发者，虽然在大多数情况下它们相同。
而如果F-Droid选择分发由应用开发者私钥签名的apk，那么注册责任就在开发者身上。虽然F-Droid声称如果构建可重现将使用原开发人员的私钥进行分发，并且F-Droid说它不能接管所分发的开源应用程序的应用程序标识符，因为这将有效地夺取这些应用程序的独家分发权。但在实际操作中态度完全相反：https://github.com/RikkaApps/Shizuku/issues/69  由于F-Droid的政策要求，我尝试性让shizuku支持了可重现构建，但是依然被F-Droid拒绝分发apk，原因是shizuku按照 Apache 2.0 第 6 节禁止了其它人使用目前的应用程序身份标识，所以shizuku不是自由软件。

</p>
</details>

> according to Apache 2.0 Section 6, so **Shizuku is not free software**.

@IzzySoft According to one of the Shizuku developer Shizuku is non-free (sorry if the translation is incorrect or I misunderstood the statement). Such apps are then allowed at IoD?

**Comment by IzzySoft** [comment]
This is quite a boarderline issue, and it's a bit unclear how to deal with it. From the Readme here:

> You are FORBIDDEN to use `manager/src/main/res/mipmap*/ic_launcher*.png` image files, unless for displaying Shizuku itself.

could be counted as NonFreeAssets.

> You are FORBIDDEN to use Shizuku as app name or use `moe.shizuku.privileged.api` as application id

should be rather clear, like a trademark protection. Even F-Droid says, if you make a fork it must have a dedicated packageName/applicationID and display name – and even icon, and thus could cover the first point as well. The problem lies in:

> or declare moe.shizuku.manager.permission.* permission.

If a fork cannot use that, what good can the fork be? That's what apps which want to use the API need to request. So forbidding that, is essentially forbidding forks that offer a similar functionality. And to my understanding, **this** part is **not** covered by section 6 of the Apache license – but I'm not a lawyer, so this is just my subjective opinion. I wish Shizuku team would drop this part. I'd say (again subjectively), there are no real issues with the other two, as outlined. Dropping this 3rd point, would IMHO be a great service to the open source community.

**Comment by GfEW** [comment]
Being not a lawyer, either, the third point sounds to me like an (possibly overreaching) attempt to apply the second point (i. e. the legit protection of the name "shizuku") to the permissions namespace.

If it's all about the name, and the code otherwise meant to be FLOSS, would it be OK to `s/shizuku/freezuku/` everywhere to create a fork that can be distributed without the limitations associated with the name "shizuku", and promote the renamed permission/API?

**Comment by IzzySoft** [comment]
> would it be OK to s/shizuku/freezuku/ everywhere to create a fork that can be distributed

It would probably be OK by the Shizuku team – but would not make much sense if the fork cannot declare the very permission apps need to request 🤷‍♂️ That's the point why I suggest to just drop that one requirement. The others are fine, and covered by section 6.

**Comment by GfEW** [comment]
>>  would it be OK to s/shizuku/freezuku/ everywhere

> would not make much sense if the fork cannot declare the very permission apps need to request

I guess I'm being naive or blind (if not both), but by _everywhere_, I mean in the API, too. 

Of course, then to
>>  promote the renamed permission/API

is easier said than done. Should it succeed however, why would apps, in order to use the `freezuku` API, need to request `shizuku` permissions anymore?

Presumably, the strong provisions regarding the original name are - at least in part - set up to protect the integrity and maintainability of the whole project, on the grounds that ill-behaving forks could introduce various sorts of trouble the shizuku team doesn't feel inclined to spawn, let alone to deal with.

The renaming approach would voluntarily trade that protection for freedom, and otherwise be bound to the original source.

That being said, I obviously agree that dropping the one requirement that (apparently) is the only remaining obstacle to free distribution would be preferable.

**Comment by IzzySoft** [comment]
I'm not an Android dev. But renaming the permission would mean all apps using it would stop working until they update their code to the new name. Providing different permissions (per "manager" app) would add complications on the end of all app developers whose projects are using the Shizuku API (which one should they request? They'd need to add code evaluating that. And what if one replaces Shizuku with a fork on the device? How would apps using the API handle that?).

It's not just a "display name" (as icon and app name would be), functionality is attached to this one.

> on the grounds that ill-behaving forks could introduce various sorts of trouble

If it's intentional ill-behaving (aka "malicious forks") – do you think those actors would be stopped by legal stuff?

But well, permit me to withdraw from this discussion. I gave my points as requested, and I'm not involved in the decision process anyway – apart from having enough other tasks on my list, sorry…

**Comment by GfEW** [comment]
> Providing different permissions (per "manager" app) would add complications on the end of all app developers whose projects are using the Shizuku API (which one should they request?

Alas, that's the "easier said than done" part._(*)_

Alternatively, `moe.shizuku.manager.permission.*` could *be declared by a separate app that translates API requests to freezuku*, and contains no actual code from shizuku.

>> on the grounds that ill-behaving forks could introduce various sorts of trouble
> If it's intentional ill-behaving (aka "malicious forks") – do you think those actors would be stopped by legal stuff?

I'm not defending shizuku's policy, but trying to make sense of it. Any experimental, unthoughtful or otherwise low-quality fork that declared `moe.shizuku.manager.permission.* permissions` would expose apps using the API to potentially undesirable consequences beyond their control. Therefore, unauthorized changes wouldn't have to be malevolent in order to be (prone to being) misattributed to "shizuku". 

> I'm not involved in the decision process anyway – apart from having enough other tasks on my list, sorry…

Understandably so! Thank you for your contributions - especially for the effort to make reproducible builds of shizuku available in the first place.

<i><details><summary>(*)</summary>However, API transitions have succeeded with other established projects, and google is forcing breaking changes upon app developers all the time.</details></i>

**Comment by thedjchi** [comment]
The ShizukuProvider (part of the Shizuku API used by apps) has a constant for Shizuku's package name, which is hard-coded and injected at build time into the library. It's not required to use the API. HOWEVER, many apps use this variable to check if Shizuku is installed before doing Shizuku things. Even if a forked app had the moe.shizuku.manager.permission.* but a different package name, some apps will think that Shizuku is not installed and will not work properly.

Fixing this would require the developers to update the library to determine the manager's package name dynamically.

Probably could be done with PackageManager.queryIntentActivities() or PackageManager.resolveActivity() checking for an intent that is unique to Shizuku, such as moe.shizuku.privileged.api.intent.action.REQUEST_PERMISSION (apps would then need to change their Manifest by adding a query for that intent filter). That way, a fork with a different package name could be recognized by an app using the Shizuku API.

**Comment by eternal-sorrow** [comment]
> Fixing this would require the developers to update the library to determine the manager's package name dynamically so that a fork with a different package name could be recognized by an app using the Shizuku API.

And then waiting for the needed (to you) apps to update for this new version of the library, which might never happen.

**Comment by Maksym-0xp** [comment]
Feel free to [Requests For Packaging](https://gitlab.com/fdroid/rfp) and see how that goes 🙃
It's not "just" releasing this app on F-Droid.

**Comment by poiNt3D** [comment]
what a moron requirement
when having multiple projects depend on you
